src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -82,13 +91,35 @@ public static function fromDsn(string $dsn, array $options = array(), bool $debu
         }
 
         $exchangeOptions = $amqpOptions['exchange'];
-        $queueOptions = $amqpOptions['queue'];
+        $queueOptions = self::normalizeQueueOptions($amqpOptions['queue']);
 
         unset($amqpOptions['queue'], $amqpOptions['exchange']);
 
         return new self($amqpOptions, $exchangeOptions, $queueOptions, $debug, $amqpFactory);
     }
 
+    private static function normalizeQueueOptions(array $queueOptions): array
+    {
+        if (!\is_array($queueOptions['arguments'] ?? false)) {
+            return $queueOptions;
+        }
+
+        foreach (self::ARGUMENTS_AS_INTEGER as $key) {
+            if (array_key_exists($key, $queueOptions['arguments'])) {
```

Comment:
[minor] You could use an early `continue` when the key does not exists to remove one condition level.